core/src/main/java/org/apache/brooklyn/core/entity/AbstractEntity.java

```diff
@@ -1548,7 +1550,7 @@ public boolean unsubscribe(Entity producer, SubscriptionHandle handle) {
      * @deprecated since 0.9.0; for internal use only
      */
     @Deprecated
-    protected synchronized SubscriptionTracker getSubscriptionTracker() {
+    protected SubscriptionTracker getSubscriptionTracker() {
```

Comment:
I think this existing code is ok, in terms of the deadlock encountered in BROOKLYN-245. The `AttributeMap` should never call into `getSubscriptionTracker()`, so there will never be a call to it while holding a lock on `AttributeMap.values`.

In terms of calling `getSubscriptionContext()` while holding other locks (e.g. on `AbstractEntity.this`) we should be ok: it will synchronize on `EntityManagementSupport.this` and will then call into the management context's subscription manager, but that code should *never* call out to alien code.

It would be good to remove the `synchronized (AbstractEntity.this)` entirely, to have simpler concurrency code. But we must be aware that the caller might already have synchronized on the entity instance when calling `subscribe()`. So removing this synchronization doesn't eliminate any deadlocks - instead we must be sure that synchronizing on the entity instance won't risk deadlocks.

As an aside, note that double-checked locking is broken if the field is not declared volatile.

Given that, are you ok leaving the code as-is in this PR?